trino-rest-github/src/main/java/pl/net/was/rest/github/Sync.java

```diff
@@ -1867,6 +1886,18 @@ private static int retryExecute(PreparedStatement statement)
                 }
                 log.severe(e.getMessage());
                 log.severe(format("Retrying %d more times", breaker));
+
+                if (queryDelay > 0) {
+                    try {
+                        MILLISECONDS.sleep(queryDelay - (System.currentTimeMillis() - startTime));
+                    }
+                    catch (InterruptedException ex) {
+                        // Do nothing
+                    }
+                }
```

Comment:
Dunno if there's a cleaner way to do a nested try-catch block like this, but simplest way I found to sleep on query failures

Reply:
We don't actually need to sleep on query failures. We can expect transient failures, like network issues. We're adding sleep after a successful query execution to avoid hitting the secondary rate limit, so we don't expect to hit it anymore. If it happens, it's already too late to sleep.

Reply:
Ah! Yup -- makes sense! Thanks for the call out 🙏